Dialog Evolution Doc Updates

Aevyz opened the pull request:
Right, hopefully I did not lose half of botboys changes like the last attempt...

Documentation builds at RD https://roboydialog.readthedocs.io/en/memory-doc/user_manual/1_getting_started.html no problem. 

#59 should be completely incorporated now, minus direct DB link.

https://github.com/Roboy/roboy_memory/pull/43

Todos:

- [x] Spell Check
- [x] Resolve Merge Conflict
- ~~[ ] Add page on expected outcomes~~
- [x] Add links (done as soon as I am on laptop)

josephbirkner committed:
Docs adjustments and fixes for NLU submodule.

Aevyz committed:
Integrate Changes into BB Dev

Aevyz committed:
[Docs] Added UML & More Updates

Aevyz committed:
[Doc] Updates

Aevyz committed:
[Docs] SM Update, Reacme and missing pic

Aevyz committed:
[Docs] Image Sources

Commit:
Merge branch 'joseph-midterm-doc-updates' of https://github.com/Roboy/roboy_dialog into memory-ref-doc

missxa commented on .gitignore:
```diff
@@ -23,3 +23,5 @@ resources/word2vec_toy_data_and_model
 
 docs/doxyxml
+
+dialog/
```
why is it ignored??

missxa commented on README.md:
```diff
@@ -1,33 +1,200 @@
+
+Yes, you can find them [here](https://roboy-dialog.readthedocs.io/).
+
+## How does the Dialog System Work
```
lets ditch this section from readme

missxa commented on README.md:
````diff
@@ -1,33 +1,200 @@
+[Set Environmental Variables](#environmental-variables). **Only do this once.**
 
-## How to run it
+```bash
````
so this is a very nice chunk of bash script, but you gotta explain what each line does: either separate it to steps or add a `#` in the end of each line with a short explanation

missxa commented on README.md:
```diff
@@ -1,33 +1,200 @@
+    - This project will run with a 4GB of RAM, however development will be a horror-show.
+- Ubuntu (or variation) 16.04 LTS
+    - Needed for `ROS Kinetic`
+    - If you do not need ROS features, any version of Linux should do
```
macOS works too

missxa commented on README.md:
```diff
@@ -1,33 +1,200 @@
+
+> Note that `apt` is bugged with Ubuntu 16.04. See: https://bugs.launchpad.net/ubuntu/+source/appstream/+bug/1583845 if you get problems with `apt-update`.
+
+### Optional
```
it's confusing that optional and Installing and Usage via Command Line are the same kind of heading. 
would make more sense to push optional and recommendations to be subsections of requirements

missxa commented on README.md:
```diff
@@ -1,33 +1,200 @@
+### Optional
+
+- ROS
+- Redis
```
not used by anyone. we shouldnt mention it

Aevyz committed:
[Docs] Removed UML Diagrams

Beyond saving... New one's shall be created by Endterm. Sources left in the Folder, to make life for future doc easier

missxa commented on README.md:
````diff
@@ -36,57 +203,52 @@ export ROS_MASTER_URI=http://rosmaster:11311
 ```
 
-#### Using the Google Word2Vec Model in NLU
+## Configuration of Roboy_Dialog
````
of roboy_dialog or Roboy Dialog

missxa commented on dialog/src/main/java/roboy/dialog/ConversationManager.java:
```diff
@@ -82,63 +82,40 @@ public static void main(String[] args) throws IOException {
 
-        //Roboy mode: Repeat a conversation a few times.
-        if(ConfigManager.ROS_ENABLED) {
+        //I/O specific core behaviour
```
are these your changes @Aevyz 
seems like botboys

Aevyz committed:
Update .gitignore

Commit:
Java version checking Hint in README

Commit:
Cleanup

josephbirkner committed:
Spelling error fix

josephbirkner committed:
Another formatting fix in README

josephbirkner committed:
Corrected openjdk-9 reference to openjdk-8

josephbirkner committed:
Another README formatting fix

josephbirkner committed:
Re-added hint about adding variable export directives to .bashrc

Aevyz committed:
Submodule Update

josephbirkner committed:
Changes from PR comments

josephbirkner committed:
Lowercase roboy_dialog

Aevyz committed:
[Docs] Update

Aevyz committed:
Merge remote-tracking branch 'origin/memory-doc' into memory-doc

missxa commented on docs/Usage/9_Troubleshooting.rst:
```diff
@@ -0,0 +1,29 @@
+Troubleshooting
```
please add a link to issues page of the github repo as:
a) a place to look for solutions as well
b) post new issues

Aevyz replied:
Is the current addendum satisfactory? Also Roboy_Parser has its issues page disabled, is that on purpose?

missxa commented on docs/Usage/0_installation.rst:
```diff
@@ -1,52 +1,26 @@
 
-Register a bot as described on the `telegram website`_.
-.._telegram website: https://core.telegram.org/bots#3-how-do-i-create-a-bot
+Optional steps
```
imo the installation of the telegram API should be moved to the tutorials and it is not an optionall installation step @SezerCan @polinominom @Mireu

Mireu replied:
Moved it to User Manual -> Configuration in b4971d9c369e22b2d2f0d77b649c34b6542891f6 .

missxa reviewed (approved):
minor fixes, otherwise good

Aevyz committed:
Update 9_Troubleshooting.rst

Aevyz committed:
Update 9_Troubleshooting.rst

Aevyz committed:
Update 9_Troubleshooting.rst

Aevyz committed:
[README] Update to new POM

Aevyz committed:
[POM] Update to Ignore Neo4J Tests

Aevyz committed:
[Bug Fix] Roboy stops refering to ROS

Previously, if ROS was down, Roboy kept complaining the he could not connect to ROS for the WildTalkState. Now instead, he simply says he's out of words.

The logic behind the choice. GNLP connects to dialog via ROS. If one were to want GNLP, one would need to enable ROS

Aevyz committed:
Updating Folder Structure

Aevyz committed:
Merge branch 'devel' into memory-doc

Aevyz committed:
ReadMe Clarifications

josephbirkner committed:
Removed PARSER_PORT reference from config docs

josephbirkner committed:
Fixed some spelling errors in README
